pkg/scanners/scannerv4/convert.go

```diff
@@ -166,44 +168,46 @@ func vulnerabilities(vulnerabilities map[string]*v4.VulnerabilityReport_Vulnerab
 	return vulns
 }
 
-func setCvss(vuln *storage.EmbeddedVulnerability, cvss *v4.VulnerabilityReport_Vulnerability_CVSS) error {
-	if cvss == nil {
+func setScoresAndScoreVersion(vuln *storage.EmbeddedVulnerability, vulnCVSS *v4.VulnerabilityReport_Vulnerability_CVSS) error {
+	if vulnCVSS == nil {
 		return nil
 	}
 	errList := errorhelpers.NewErrorList("failed to parse vector")
-	if v2 := cvss.GetV2(); v2 != nil {
+	if v2 := vulnCVSS.GetV2(); v2 != nil {
 		if c, err := cvssv2.ParseCVSSV2(v2.GetVector()); err == nil {
 			err = cvssv2.CalculateScores(c)
 			if err != nil {
 				errList.AddError(fmt.Errorf("calculating CVSS v2 scores: %w", err))
 			}
 			// Use the report's score if it exists.
-			if v2.GetBaseScore() != 0.0 {
+			if baseScore := v2.GetBaseScore(); baseScore != 0.0 && baseScore != c.Score {
+				log.Debugf("Calculated CVSSv2 score does not match given base score (%f != %f) for %s. Using given score...", c.Score, baseScore, vuln.GetCve())
 				c.Score = v2.GetBaseScore()
 			}
 			c.Severity = cvssv2.Severity(c.Score)
 			vuln.CvssV2 = c
-			// This sets the top level score for use in policies. It will be overwritten if
-			// v3 exists.
+			// This sets the top level score for use in policies.
+			// It will be overwritten if v3 exists.
```

Comment:
Curious: Feels like it's your preference; why change?

Reply:
Just seemed odd to have one super tiny line instead of two almost similarly-sized full sentences

Reply:
I see comments as paragraphs.  They are broken at a certain column for readability, similar to how markdown and a bunch of other text-based markups do. The funny thing is that one sentence per line is sometimes a style guide for markdown and others. So, I was curious.